rasa/core/policies/unexpected_intent_policy.py

```diff
@@ -0,0 +1,914 @@
+
+        Args:
+            model_path: Path where model is to be persisted.
+        """
+        model_utilties = super()._load_model_utilities(model_path)
+        label_quantiles = io_utils.pickle_load(
+            model_path / f"{cls._metadata_filename()}.label_quantiles.pkl"
+        )
+        model_utilties.update({"label_quantiles": label_quantiles})
+        return model_utilties
+
+    @classmethod
+    def _update_loaded_params(cls, meta: Dict[Text, Any]) -> Dict[Text, Any]:
+        meta = train_utils.override_defaults(cls.defaults, meta)
+        return meta
+
+    @classmethod
+    def _load_policy_with_model(
+        cls,
+        model: "IntentTED",
+        featurizer: TrackerFeaturizer,
+        model_utilities: Dict[Text, Any],
+        should_finetune: bool,
+    ) -> "UnexpecTEDIntentPolicy":
+        return cls(
+            featurizer=featurizer,
+            priority=model_utilities["priority"],
+            model=model,
+            fake_features=model_utilities["fake_features"],
+            entity_tag_specs=model_utilities["entity_tag_specs"],
+            should_finetune=should_finetune,
+            label_quantiles=model_utilities["label_quantiles"],
+            **model_utilities["meta"],
+        )
+
+
+class IntentTED(TED):
+    """Follows TED's model architecture from https://arxiv.org/abs/1910.00486.
+
+    However, it has been re-purposed to predict multiple
+    labels (intents) instead of a single label (action).
+    """
+
+    def _prepare_label_classification_layers(self, predictor_attribute: Text) -> None:
+        """Prepares layers & loss for the final label prediction step."""
+        self._prepare_embed_layers(predictor_attribute)
+        self._prepare_embed_layers(LABEL)
+        self._prepare_dot_product_loss(LABEL, self.config[SCALE_LOSS])
+
+    def _prepare_dot_product_loss(
+        self, name: Text, scale_loss: bool, prefix: Text = "loss",
+    ) -> None:
+        self._tf_layers[f"{prefix}.{name}"] = self.dot_product_loss_layer(
+            self.config[NUM_NEG],
+            scale_loss,
+            similarity_type=self.config[SIMILARITY_TYPE],
+        )
+
+    @property
+    def dot_product_loss_layer(self) -> tf.keras.layers.Layer:
+        """Returns the dot-product loss layer to use.
+
+        Multiple intents can be valid simultaneously, so `IntentTED` uses the
+        `MultiLabelDotProductLoss`.
+
+        Returns:
+            The loss layer that is used by `_prepare_dot_product_loss`.
+        """
+        return layers.MultiLabelDotProductLoss
+
+    @staticmethod
+    def _get_labels_embed(
+        label_ids: tf.Tensor, all_labels_embed: tf.Tensor
+    ) -> tf.Tensor:
+        # instead of processing labels again, gather embeddings from
+        # all_labels_embed using label ids
+
+        indices = tf.cast(label_ids[:, :, 0], tf.int32)
+
+        # Find padding indices. They should have a value -1
+        padding_indices = tf.where(tf.equal(indices, -1))
```

Comment:
We should use the `LABEL_PAD_ID` here.
```suggestion
        # Find padding indices. They should have a value `LABEL_PAD_ID`
        padding_indices = tf.where(tf.equal(indices, LABEL_PAD_ID))
```

Reply:
Same throughout code.